packages/next/src/server/after/after-context.ts

```diff
@@ -98,19 +94,27 @@ export class AfterContext {
 
   private async runCallbacksOnClose() {
     await new Promise<void>((resolve) => this.onClose!(resolve))
-    return this.runCallbacks(this.requestStore!)
+    return this.runCallbacks(this.requestStore)
   }
 
-  private async runCallbacks(requestStore: RequestStore): Promise<void> {
+  private async runCallbacks(
+    requestStore: undefined | RequestStore
+  ): Promise<void> {
     if (this.callbackQueue.size === 0) return
 
-    const readonlyRequestStore: RequestStore =
-      wrapRequestStoreForAfterCallbacks(requestStore)
+    const readonlyRequestStore: undefined | RequestStore =
+      requestStore === undefined
+        ? undefined
+        : // TODO: This is not sufficient. It should just be the same store that mutates.
+          wrapRequestStoreForAfterCallbacks(requestStore)
```

Comment:
@lubieowoce Instead of wrapping, we should mutate something in the request store when we change scope to handle the case where it's started earlier.